### PR TITLE
Turn on language-service integration rewrite 

### PR DESCRIPTION
**Pull request by davkean**
This merges the language-service integration rewrite into master and makes it replace the existing integration. All of these changes have already been reviewed or are out for review, with exception to https://github.com/dotnet/project-system/pull/4419/commits/e4b0c17eb2392d19ff823d15f0ea2f41acdacbc5 and https://github.com/dotnet/project-system/pull/4419/commits/a7e646f9f62b92435c1b1393dac5f6a506dca47d.

This fixes: https://github.com/dotnet/project-system/issues/2733.

Currently broken in Preview 2 bits (unrelated to this change):

### Roslyn
[Roslyn leaking AbstractProjects](https://github.com/dotnet/roslyn/issues/32036)
[Project Context switch icon changes when moving caret](https://github.com/dotnet/roslyn/issues/28236)
[Multi-targeting context is broken in 16.0 Preview 1](https://github.com/dotnet/roslyn/issues/31365)
[With "Full Solution Analysis" turned on, certain syntax errors get duplicated](https://github.com/dotnet/roslyn/issues/28808)
[Analyzers warning/errors/messages are not correctly grouped for multi-targeting projects](https://github.com/dotnet/roslyn/issues/32055)

### Project System
[Build warnings & errors from build always get applied to the first TFM in the project](https://github.com/dotnet/project-system/issues/4034)

### Performance
- [x] Solution-load time loading Roslyn.sln
- [x] Solution-load time loading ProjectSystem.sln
- [x] Solution-load time File -> New -> Console App

### Error List
- [x] Errors from csc/vbc/fsc during build end up in Error List (single-TFM)
- [x] Errors from csc/vbc/fsc during build end up in Error List (multi-TFM) (broken due to https://github.com/dotnet/project-system/issues/4034)
- [x] Errors from background analysis end up in Error List (single-TFM)
- [x] Errors from background analysis end up in Error List (multi-TFM)

### Project manipulation
- [x] Add/remove a reference and Roslyn sees change (single-TFM)
- [x] Add/remove a reference and Roslyn sees change (multi-TFM)
- [x] Add/remove a source file and Roslyn sees change (single-TFM)
- [x] Add/remove a source file and Roslyn sees change (multi-TFM)
- [x] Add/remove a Razor file and Roslyn sees change (single-TFM)
- [x] Add/remove an analyzer and Roslyn sees change (single-TFM)
- [x] Add/remove an analyzer and Roslyn sees change (multi-TFM)
- [x] Add/remove an additional file and Roslyn sees change (single-TFM)
- [x] Add/remove an additional file and Roslyn sees change (multi-TFM)
- [x] Change conditional compilation symbols and Roslyn sees change (single-TFM)
- [x] Change conditional compilation symbols and Roslyn sees change (multi-TFM)
- [x] Retarget project from one TFM -> another
- [x] Change configuration (single-TFM) [note, this and next scenario are affected by https://github.com/dotnet/roslyn/issues/32036]
- [x] Change configuration (multi-TFM)

### Features
- [ ] Razor designer works (broken by https://github.com/dotnet/roslyn/issues/31365)
- [x] Edit & Continue works
- [x] CodeLens

### CODE REVIEW

**Comment by davkean** [comment]
This is blocked by https://github.com/dotnet/roslyn/issues/31365.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs [hunk 1]
```diff
@@ -83,5 +89,24 @@ public void UnregisterContext(IWorkspaceProjectContext context)
...
+        private string GetWorkspaceContextIdFromActiveContext()
+        {
+
+            return _threadingService.ExecuteSynchronously(() =>
+            {
+                try
+                {
+                    // If we're never been set an active context, we just
+                    // pick one based on the active configuration.
+                    return _activeWorkspaceProjectContextHost.OpenContextForWriteAsync(a => Task.FromResult(a.ContextId));
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs]: This needs `await` in order to observe the exception, right? Or is `OperationCanceledException` special?

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs]: > Or is OperationCanceledException special

Nope, good catch.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs]:
```suggestion
                    return await _activeWorkspaceProjectContextHost.OpenContextForWriteAsync(a => Task.FromResult(a.ContextId));
```

**Comment by jmarolf** [comment]
@davkean 	I assume you are asleep but I plan to merge this in the next 15 minutes unless I hear from you

**Pull request merged**